Index: libgc/include/private/gcconfig.h
===================================================================
--- libgc.orig/include/private/gcconfig.h
+++ libgc/include/private/gcconfig.h
@@ -310,6 +310,10 @@
 #    define SPARC
 #    define mach_type_known
 # endif
+# if defined(LINUX) && defined(__avr32__)
+#    define AVR32
+#    define mach_type_known
+# endif
 # if defined(LINUX) && defined(__sh__)
 #    define SH
 #    define mach_type_known
@@ -2115,6 +2119,19 @@
 #   endif
 #endif
 
+# ifdef AVR32
+#   define MACH_TYPE "AVR32"
+#   define CPP_WORDSZ 32
+#   define ALIGNMENT 4
+#   define OS_TYPE "LINUX"
+#   define DYNAMIC_LOADING
+#   define LINUX_STACKBOTTOM
+#   define USE_GENERIC_PUSH_REGS
+#   define SEARCH_FOR_DATA_START
+      extern int _end[];
+#   define DATAEND (_end)
+# endif
+
 # ifdef CRIS
 #   define MACH_TYPE "CRIS"
 #   define CPP_WORDSZ 32
