Index: libgc/configure.ac
===================================================================
--- libgc.orig/configure.ac
+++ libgc/configure.ac
@@ -572,6 +572,9 @@ case "$host" in
     dnl This is now redundant, but it is also important for incremental GC
     dnl performance under Irix.
     ;;
+ sparc*-*-linux*)
+    machdep="mach_dep.lo sparc_mach_dep.lo"
+    ;;
  sparc-*-netbsd*)
     machdep="sparc_netbsd_mach_dep.lo"
     ;;
