Index: b/include/gc_config_macros.h
===================================================================
--- a/include/gc_config_macros.h
+++ b/include/gc_config_macros.h
@@ -125,6 +125,10 @@
 # if defined(__rtems__) && (defined(i386) || defined(__i386__))
 #   define GC_RTEMS_PTHREADS
 # endif
+# if defined(__gnu_hurd__)
+#   define GC_GNU_THREADS
+#   define GC_PTHREADS
+# endif
 #endif /* GC_THREADS */
 
 #undef GC_PTHREADS
