Index: b/include/gc_config_macros.h
===================================================================
--- a/include/gc_config_macros.h
+++ b/include/gc_config_macros.h
@@ -99,7 +99,7 @@
        && !defined(GC_OSF1_THREADS) && !defined(GC_IRIX_THREADS)
     /* FIXME: Should we really need for FreeBSD and NetBSD to check     */
     /* that no other GC_xxx_THREADS macro is set?                       */
-#   if defined(__FreeBSD__) || defined(__DragonFly__)
+#   if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 #     define GC_FREEBSD_THREADS
 #   elif defined(__NetBSD__)
 #     define GC_NETBSD_THREADS
@@ -121,6 +121,10 @@
 # if defined(__rtems__) && (defined(i386) || defined(__i386__))
 #   define GC_RTEMS_PTHREADS
 # endif
+# if defined(__gnu_hurd__)
+#   define GC_GNU_THREADS
+#   define GC_PTHREADS
+# endif
 #endif /* GC_THREADS */
 
 #undef GC_PTHREADS
